src/main/java/org/jenkinsci/plugins/workflow/support/actions/LogActionImpl.java

```diff
@@ -110,8 +101,8 @@
     private String charset;
 
     private LogActionImpl(FlowNode parent, Charset charset) {
-        if (!isRunning(parent)) {
-            throw new IllegalStateException("cannot start writing logs to a finished node " + parent);
+        if (!parent.isActive()) {
+            LOGGER.log(Level.WARNING, null, new IllegalStateException("cannot start writing logs to a finished node " + parent + " " + parent.getDisplayFunctionName() + " in " + parent.getExecution()));
```

Comment:
The exception is not “swallowed”, this is just a decision to log a stack trace.

My reasoning for switching from throwing an exception to just logging a warning was that an earlier mistake in the `isActive` implementation showed that the build could be completely broken by such a bug, which seems too harsh given that there is no real harm done if some content happens to be written to a step log after the step is supposedly complete. It may not show up in the build log but that is all. We do want to fix such bugs, hence the stack trace for diagnosis.